CONTRIBUTING.md

```diff
@@ -97,46 +97,33 @@ git fetch --all
 
 # Create the new release branch from the one on crossplane/crossplane.
 git checkout -b $RELEASE_BRANCH upstream/$RELEASE_BRANCH
+git reset --hard vX.Y.0 # or whatever version we are releasing
```

Comment:
It's only happened to me a few times, but I _have_ accidentally done a hard reset on code I hadn't yet pushed. Could we suggest checking out at that tag rather to avoid that possibility?